packages/ui/src/components/sign-up/sign-up.tsx

```diff
@@ -1,34 +1,51 @@
 import { useClerk } from '@clerk/clerk-react';
 import * as Common from '@clerk/elements/common';
 import * as SignUp from '@clerk/elements/sign-up';
+import type { EnvironmentResource } from '@clerk/types';
 
 import { EmailField } from '~/common/email-field';
 import { FirstNameField } from '~/common/first-name-field';
 import { LastNameField } from '~/common/last-name-field';
 import { OTPField } from '~/common/otp-field';
 import { PasswordField } from '~/common/password-field';
+import { PhoneNumberField } from '~/common/phone-number-field';
+import { UsernameField } from '~/common/username-field';
 import { PROVIDERS } from '~/constants';
-import { getEnabledSocialConnectionsFromEnvironment } from '~/hooks/getEnabledSocialConnectionsFromEnvironment';
 import { Button } from '~/primitives/button';
 import * as Card from '~/primitives/card';
 import * as Connection from '~/primitives/connection';
 import * as Icon from '~/primitives/icon';
 import { Seperator } from '~/primitives/seperator';
+import { TextButton } from '~/primitives/text-button';
+import { getEnabledSocialConnectionsFromEnvironment } from '~/utils/getEnabledSocialConnectionsFromEnvironment';
 
 export function SignUpComponent() {
-  const clerk = useClerk();
-  // @ts-ignore clerk is not null
-  const enabledConnections = getEnabledSocialConnectionsFromEnvironment(clerk?.__unstable__environment);
-
   return (
     <SignUp.Root>
-      <Common.Loading>
-        {isGlobalLoading => {
-          return (
-            <SignUp.Step
-              data-clerk
-              name='start'
-            >
+      <SignUpComponentLoaded />
```

Comment:
Move to loaded component so we can know that clerk has loaded via elements `<SignUp.Root>` to interface with useClerk hook